moz-webgpu-cts/src/main.rs

```diff
@@ -125,6 +190,474 @@ fn run(cli: Cli) -> ExitCode {
+                        }
+                    })
+                    .collect::<Vec<_>>();
+
+                if found_glob_parse_err {
+                    log::error!("failed to parse one or more WPT report globs; bailing");
+                    return ExitCode::FAILURE;
+                }
+
+                globs
+            };
+
+            if report_paths.is_empty() && report_globs.is_empty() {
+                log::error!("no report paths specified, bailing");
+                return ExitCode::FAILURE;
+            }
+
+            let exec_report_paths = {
+                let mut found_glob_walk_err = false;
+                let files = report_globs
+                    .iter()
+                    .flat_map(|(base_path, glob)| {
+                        glob.walk(base_path)
+                            .filter_map(|entry| match entry {
+                                // TODO: warn when not ending in `wptreport.json`
+                                Ok(entry) => Some(entry.into_path()),
+                                Err(e) => {
+                                    found_glob_walk_err = true;
+                                    let ctx_msg = if let Some(path) = e.path() {
+                                        format!(
+                                            "failed to enumerate files for glob `{}` at path {}",
+                                            glob,
+                                            path.display()
+                                        )
+                                    } else {
+                                        format!("failed to enumerate files for glob `{glob}`")
+                                    };
+                                    let e = Report::msg(e).wrap_err(ctx_msg);
+                                    eprintln!("{e:?}");
+                                    None
+                                }
+                            })
+                            .collect::<Vec<_>>() // OPT: Can we get rid of this somehow?
+                    })
+                    .chain(report_paths)
+                    .collect::<Vec<_>>();
+
+                if found_glob_walk_err {
+                    log::error!("failed to enumerate files with WPT report globs; bailing");
+                    return ExitCode::FAILURE;
+                }
+
+                files
+            };
+
+            if exec_report_paths.is_empty() {
+                log::error!("no WPT report files found, bailing");
+                return ExitCode::FAILURE;
+            }
+
+            log::trace!("working with the following WPT report files: {exec_report_paths:#?}");
+            log::info!("working with {} WPT report files", exec_report_paths.len());
+
+            // TODO: If we don't need to consult metadata (because we plan on ignoring everything
+            // there, and deleting anything not in reports), then don't even load these.
```

Comment:
- [ ] Avoid loading metadata when `--preset=reset-all` is specified.

Reply:
Actually, broke this out to https://github.com/ErichDonGubler/moz-webgpu-cts/issues/50.

Reply:
d4dade918aee50e1560ce06d6a89a68090dad982